### ISSUE TITLE
Some time the response "just hangs.."

### ISSUE THREAD

**Issue by AlmogBaku** [elazarl/goproxy#178]
I don't know why yet, but sometime when I'm doing a request (specifically to this url) it just hangs. 
Either from browser or curl:

```
curl --proxy-user user:pass --proxy myproxy.com:8080 http://m.booking.com/
```

Proxy logs:

```
2016/08/10 09:22:16 [110] INFO: Got request / m.booking.com GET http://m.booking.com/
2016/08/10 09:22:16 [110] INFO: Sending request GET http://m.booking.com/
```

Useless to say that without proxy it works as a charm.. (and booking.com is big enough to assume their servers probably well)

**Comment by elazarl** [comment]
Hmmm...

many thanks for the report.

I won't have time to investigate it in the next few weeks, but can you
attach a pcap, both of client->proxy and proxy->booking.com

**Comment by AlmogBaku** [comment]
[http-booking-extracted.pcap.zip](https://github.com/elazarl/goproxy/files/412659/http-booking-extracted.pcap.zip)

**Comment by AlmogBaku** [comment]
It seems from there that the proxy keep the connection open until the end-server just stop to answer..
https://www.w3.org/Protocols/HTTP/Issues/http-persist.html

**Comment by AlmogBaku** [comment]
Squid define a timeout for idle connections to the servers: http://www.squid-cache.org/Versions/v3/3.1/cfgman/pconn_timeout.html

**Comment by elazarl** [comment]
In principle all outbound connection goes through http.Transport, so it
should manage idle connection pools, timeouts, and the like. And of course
inbound connections by Go's http.

So I'm not sure we need anything specific.

**Comment by AlmogBaku** [comment]
https://github.com/golang/go/issues/13998

**Comment by AlmogBaku** [comment]
seems like it should be fixed in 1.7+
